src/Tests/dotnet-new.Tests/CapabilityExpressionEvaluationTests.cs

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#nullable enable
+
+using FluentAssertions;
+using Microsoft.NET.TestFramework;
+using Microsoft.NET.TestFramework.Commands;
+using Xunit;
+using Xunit.Abstractions;
+using Microsoft.NET.TestFramework.Assertions;
+using System.IO;
+using System.Collections.Generic;
+using Microsoft.TemplateEngine.MSBuildEvaluation;
+using System;
+
+namespace Microsoft.DotNet.New.Tests
+{
+    public class CapabilityExpressionEvaluationTests 
+    {
+        [Theory]
+        [InlineData("Capability", "", false)]
+        [InlineData("Cap1", "cap1", true)]
+        [InlineData("Cap1 | Cap2", "Cap3", false)]
+        [InlineData("Cap1 | Cap2", "Cap1", true)]
+        [InlineData("Cap1 & Cap2", "Cap3", false)]
+        [InlineData("Cap1 & Cap2", "Cap1", false)]
+        [InlineData("Cap1 & Cap2", "Cap1|Cap2", true)]
+        [InlineData("Cap1 + Cap2", "Cap3", false)]
+        [InlineData("Cap1 + Cap2", "Cap1", false)]
+        [InlineData("Cap1 + Cap2", "Cap1|Cap2", true)]
+        [InlineData("!Cap3", "Cap3", false)]
+        [InlineData("!Cap3", "Cap1", true)]
+        [InlineData("(Cap1 | Cap2) + (Cap3 | Cap4)", "Cap3", false)]
+        [InlineData("(Cap1 | Cap2) + (Cap3 | Cap4)", "Cap3|Cap1", true)]
+        [InlineData("(Cap1 | Cap2) | (Cap3 | Cap4)", "Cap3", true)]
```

Comment:
I'd add an explicit test for correct `&` and `|` precedence when presented together without parentheses. This is easy to break and hard to spot from the code if it's behaving correctly